core/vm/instructions_zkevm.go

```diff
@@ -161,11 +163,77 @@ func makeLog_zkevm(size int) executionFunc {
 
 		d := scope.Memory.GetCopy(int64(mStart.Uint64()), int64(mSize.Uint64()))
 
-		// [zkEvm] fill 0 at the end
-		dataLen := len(d)
-		lenMod32 := dataLen & 31
-		if lenMod32 != 0 {
-			d = append(d, make([]byte, 32-lenMod32)...)
+		forkBlock := uint64(0)
+		if interpreter.evm.ChainConfig().ForkID88ElderberryBlock != nil {
+			forkBlock = interpreter.VM.evm.ChainConfig().ForkID88ElderberryBlock.Uint64()
+		}
+		blockNo := interpreter.VM.evm.Context().BlockNumber
+
+		// APPLY BUG ONLY ABOVE FORKID9
+		if forkBlock == 0 || blockNo < forkBlock {
```

Comment:
It seems like the block range is [fork4, fork8), excluding fork8 and beyond?

Reply:
Yeah my bad this is messy logic - but it applies the bug where block is above forkid8